src/contracts/core/DelegationManager.sol

```diff
@@ -732,6 +751,32 @@ contract DelegationManager is
         _beaconChainSlashingFactor[staker] = bsf;
     }
 
+    /// @dev Calculate amount of slashable shares that are queued withdrawn from an operator for a strategy.
+    function _getSlashableSharesInQueue(
+        address operator,
+        IStrategy strategy,
+        uint64 maxMagnitude
+    ) internal view returns (uint256) {
+        uint256 currCumulativeScaledShares = uint256(_cumulativeScaledSharesHistory[operator][strategy].latest());
+        // Note: this will simply return 0 if no history exists, same for latest() as well
+        uint256 pastCumulativeScaledShares = _cumulativeScaledSharesHistory[operator][strategy].upperLookupRecent(
+            block.number - MIN_WITHDRAWAL_DELAY_BLOCKS
+        );
```

Comment:
The completable block for a withdrawal and the block we use for the scaled share history lookup are the same block. 

There's a racing condition where i have a withdrawal that can be completed in the same block a slash occurs - if my withdrawal is completed first, then we burn too many shares when the slash lands.

Can you confirm this is an issue? If so, we need a patch + regression test.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/881/commits/4799402ad08aacd77cf121280bb2e91e62e5dfe8#diff-b8d08460d9bf99760a27f1f51eb20e5293abe371cc45d3442b62180da942bd0aR4279

Tested to be working as intended here. Checks the following:

1. On the block right before the withdrawal is completable, attempting to complete withdrawal reverts and the slashable shares are in the queue.
2. On the block the withdrawal is completable, completing a withdrawal doesn't revert and the slashable shares in the queue are 0.
3. Burning shares after step 2: results in 0 shares burned since operatorShares are 0 and all the staker's shares were withdrawn and completed.

Reply:
Thanks for the test. I think this isn't a race condition because our `upperLookup` library methods only return if a key is strictly greater than the input key.